### PR TITLE
Physics update in cumulus convection, PBL & surface layer for UFS_P7

### PR DESCRIPTION
**Pull request by JongilHan66**
## Description

(Instructions: this, and all subsequent sections of text should be removed and filled in as appropriate.)
Provide a detailed description of what this PR does.  
What bug does it fix, or what feature does it add?  
Is a change of answers expected from this PR?  

Cumulus convection:
a) More strict convection trigger
b) Reduced entrainment rate below cloud base
c) Enhanced downdraft detrainments starting from 60mb above the ground surface
d) Reduced rain evaporation
e) Modification of cloud depth separating shallow convection from deep convection

PBL and surface layer:
a) Inclusion of wind shear effect reducing characteristic mixing length
b) Reduction of background diffusivity in the inversion layers as a function of surface roughness and green vegetation fraction
c) PBL updraft overshooting limited by bulk Richardson number-based-PBL depth
d) Inclusion of new canopy heat storage parameterization
e) Modification of thermal roughness calculation over land
f) Increase of momentum roughness length over sea
g) Inclusion of sea spray effect parameterization

The verification results for the physics updates above can be found in https://www.emc.ncep.noaa.gov/gmb/jhan/vsdbw/ccaa14
More detailed description for the updates and ccaa14 results are given in the attached ppt file.
[Tests_physics_MRF_MJO_JHan_WLi.pptx](https://github.com/NOAA-EMC/fv3atm/files/6532234/Tests_physics_MRF_MJO_JHan_WLi.pptx)

Verification results for additional physics updates are in:
1. Increased sea spray effects: https://www.emc.ncep.noaa.gov/gmb/jhan/vsdbw/ccaa27
2. Inclusion of new maximum z/L parameterization in the GFS surface layer scheme:
    Winter: https://www.emc.ncep.noaa.gov/gmb/jhan/vsdbw/ccaa41
   Summer: https://www.emc.ncep.noaa.gov/gmb/jhan/vsdbw/ccaa41x

The final update is based on 'ccaa41'.

Description for the new maximum z/L parameterization can be found in the attached ppt file.
[PBL_SL_inv_JHan.pptx](https://github.com/NOAA-EMC/fv3atm/files/6532241/PBL_SL_inv_JHan.pptx)

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
- fixes #<issue_number>
- fixes noaa-emc/fv3atm/issues/<issue_number>

## Testing

How were these changes tested?  
What compilers / HPCs was it tested with?  
Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)  
Have the ufs-weather-model regression test been run? On what platform?  
- Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
- Please commit the regression test log files in your ufs-weather-model branch

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs

https://github.com/NCAR/ccpp-physics/pull/665
https://github.com/ufs-community/ufs-weather-model/pull/589

### CODE REVIEW

File: ccpp/data/GFS_typedefs.F90 [hunk 1]
```diff
@@ -3339,11 +3339,13 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
...
+!   real(kind=kind_phys) :: clam_deep      = 0.1             !< c_e for deep convection (Han and Pan, 2011, eq(6))
+    real(kind=kind_phys) :: clam_deep      = 0.07            !< c_e for deep convection (Han and Pan, 2011, eq(6))
     real(kind=kind_phys) :: c0s_deep       = 0.002           !< convective rain conversion parameter
     real(kind=kind_phys) :: c1_deep        = 0.002           !< conversion parameter of detrainment from liquid water into grid-scale cloud water
-    real(kind=kind_phys) :: betal_deep     = 0.05            !< fraction factor of downdraft air mass reaching ground surface over land
-    real(kind=kind_phys) :: betas_deep     = 0.05            !< fraction factor of downdraft air mass reaching ground surface over sea
+    real(kind=kind_phys) :: betal_deep     = 0.01            !< fraction factor of downdraft air mass reaching ground surface over land
+    real(kind=kind_phys) :: betas_deep     = 0.01            !< fraction factor of downdraft air mass reaching ground surface over sea
+    real(kind=kind_phys) :: evef           = 0.09            !< evaporation factor from convective rain
```
Comment (junwang-noaa) [inline ccpp/data/GFS_typedefs.F90:3362]: what is the difference between evef and evfact_deep? Both have same documentation.

Comment (JongilHan66) [reply ccpp/data/GFS_typedefs.F90:3362]: evfact_deep & evfactl_deep were commented out in SAS deep convection scheme, are no longer used, and will be eventually removed from both convection schemes and GFS_typedefs.F90.

File: .gitmodules [hunk 2]
```diff
@@ -5,8 +5,8 @@
 [submodule "ccpp/framework"]
 	path = ccpp/framework
 	url = https://github.com/NCAR/ccpp-framework
-	branch = main
+	branch = master
```
Comment (climbfuji) [inline .gitmodules]: this must be `main`

Comment (junwang-noaa) [reply .gitmodules]: Jongil, would you please fix this?

Comment (JongilHan66) [reply .gitmodules]: I don't understand what's the problem and please direct me to fix it.

Comment (junwang-noaa) [reply .gitmodules]: In the file .gitmodules in your fv3 branch han_p7, you are using "master" branch for ccpp-framework, please change it back to "main".

Comment (JongilHan66) [reply .gitmodules]: In the file .gitmodules in my fv3 branch han_p7, I changed "master" to "main" and commit and pushed back. Should I also update my ufs-weather-model branch (ufs_hanp7)? Also, do I need to re-do the RT run with this ccpp-framework?

Comment (climbfuji) [reply .gitmodules]: No, because the hash of ccpp-framework that fv3atm uses is still the same.

**Review by junwang-noaa — APPROVED** [review]

**Review by climbfuji — APPROVED** [review]

**Comment by climbfuji** [comment]
Submodule pointer for ccpp-physics is correct, and `.gitmodules` is back to what it should be. This PR can be merged (I cannot). @DusanJovic-NOAA @junwang-noaa

**Pull request merged**